### PR TITLE
Automate encryption using GCP KMS API

### PR DESCRIPTION
**Pull request by Edwin-Pau**
When using KMS encryption for the secrets inside the deployments'
tfvars files, users can now run a python script which will automate
the creation of the key ring, crypto key, and the encryption of the
tfvars secrets using the newly created crypto key.

Signed-off-by: Edwin Pau <epau@teradici.com>

### CODE REVIEW

File: tools/gcp_kms_encrypt_secrets.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            
+    except Exception as err:
+        print("An exception occurred encrypting secrets:")
+        print(err)
+        print("")
+      
+
+# Entry point to this script
+if __name__ == '__main__':
+    main()
```
Comment (shermanyin) [inline tools/gcp_kms_encrypt_secrets.py:213]: newline at end of file please.

File: tools/gcp_kms_encrypt_secrets.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
+# Copyright (c) 2020 Teradici Corporation
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+#!/usr/local/bin/python3
+
+from google.cloud import kms_v1
```
Comment (shermanyin) [inline tools/gcp_kms_encrypt_secrets.py:8]: alphabetical order imports

File: tools/gcp_kms_encrypt_secrets.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        with open("{}.encrypted".format(cam_credentials_file), "w") as cam_credentials_encrypted_file:
+            cam_credentials_encrypted_file.write(cam_credentials_encrypted)
+            
+    except Exception as err:
+        print("An exception occurred encrypting CAM JSON credentials:")
+        print(err)
+        print("")
+
+
+# Driver of this script
```
Comment (shermanyin) [inline tools/gcp_kms_encrypt_secrets.py:151]: this comment is not needed

**Review by shermanyin — CHANGES_REQUESTED** [review]
hey I just took a one minute look at the code. I think it's a great idea to have this helper script. 
- could you please include instructions in the appropriate section of the readme file on how to use this? you should offer 2 options: 1) for the experienced, just tell them to include encrypted strings in the tfvars. 2) for the easy way, run this python script
- instead of asking users to edit this file, you should make user either enter the parameters like project-id, credentials-file, ring_id, etc in run time (e.g. interactively on the CLI) or in a config file.  
- or better yet, maybe we can ask users to enter the plain text secrets in the tvfars file, then run this script and this script will parse the tfvars for things it needs like project-id, then overwrite the file with encrypted secrets where needed.  so this way, the user enters info only in one place (tfvars), run this script ON that file which encrypts the secrets in place, then they have a tfvars that they can directly use for the deployment.

**Comment by Edwin-Pau** [comment]
Noted! Will make the appropriate changes and create a new pull request when ready.

**Comment by Edwin-Pau** [comment]
A new branch (gcp-kms-script) and pull requested is created to better reflect the functionality of the script since it now encrypts and decrypts. New pull request will reflect code changes made per the comments here.

**Pull request closed without merging**